From: David Sterba <dsterba@suse.cz>
Date: Fri, 22 Feb 2013 10:50:30 +0100
Patch-mainline: no
Subject: [PATCH] btrfs: fixup writeback_inodes_sb wrappers use

Fix build failure.  Use the available exported wrapper
try_to_writeback_inodes_sb_nr that already exists.

Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/extent-tree.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fs/btrfs/extent-tree.c b/fs/btrfs/extent-tree.c
index 27c8615..ff30740 100644
--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -3792,6 +3792,8 @@ static int can_overcommit(struct btrfs_root *root,
 static inline int writeback_inodes_sb_nr_if_idle_safe(struct super_block *sb,
 					       unsigned long nr_pages)
 {
+/* use try_to_writeback_inodes_sb_nr instead, it's equivalent but exported */
+#if 0
 	/* the flusher is dealing with the dirty inodes now. */
 	if (writeback_in_progress(sb->s_bdi))
 		return 1;
@@ -3802,6 +3804,7 @@ static inline int writeback_inodes_sb_nr_if_idle_safe(struct super_block *sb,
 		return 1;
 	}
 
+#endif
 	return 0;
 }
 
@@ -3812,6 +3815,7 @@ void btrfs_writeback_inodes_sb_nr(struct btrfs_root *root,
 	int started;
 
 	/* If we can not start writeback, just sync all the delalloc file. */
+	/* started = writeback_inodes_sb_nr_if_idle_safe(sb, nr_pages); */
 	started = writeback_inodes_sb_nr_if_idle_safe(sb, nr_pages);
 	if (!started) {
 		/*
-- 
1.8.1.2

